sphinx/themes/basic/layout.html

```diff
@@ -103,7 +103,25 @@
     <link rel="stylesheet" href="{{ pathto('_static/' + style, 1) }}" type="text/css" />
     <link rel="stylesheet" href="{{ pathto('_static/pygments.css', 1) }}" type="text/css" />
     {%- for cssfile in css_files %}
-    <link rel="stylesheet" href="{{ pathto(cssfile, 1) }}" type="text/css" />
+        {%- if cssfile.filename is defined %}
+            {%- set filename = cssfile.filename %}
+        {%- else %}
+            {%- set filename = cssfile %}
+        {%- endif %}
+        {%- if cssfile.alternate is defined %}
+            {%- set alt = cssfile.alternate %}
+        {%- endif %}
+        {%- if cssfile.title is defined and cssfile.title is string %}
+            {%- if cssfile.alternate is not defined %}
+                {%- set alt = True %}
+            {%- endif %}
+    <link rel="{% if alt %}alternate {% endif %}stylesheet" href="{{ pathto(filename, 1) }}" type="text/css" title="{{ cssfile.title }}" />
+        {%- else %}
+            {%- if cssfile.alternate is not defined %}
+                {%- set alt = False %}
+            {%- endif %}
+    <link rel="{% if alt %}alternate {% endif %}stylesheet" href="{{ pathto(filename, 1) }}" type="text/css" />
+        {%- endif %}
```

Comment:
But it has to be in the template if I want to allow css_files being modified in a template file (see tests/roots/test-stylesheets/_templates/layout.html). Or can I process css_files between its definition with:

    {% set css_files = ... %}

and its implicit use with:

    {{ css() }}

?

The problem is that the default value for "alternate", if it's not specified, depends on whether or not a "title" attribute was provided. I can easily handle this on the python side (in add_stylesheet), but that's not the only way to add css files.